### PR TITLE
Duoswap on ALGO

### PR DESCRIPTION
**Pull request by chrisnevers**
* Have accounts accept non-network tokens in frontend.
* Adjust ALGO runtime shim, so calling `getViews` will call `delay` instead of `not_yet`.
* Fix overflow on ALGO when performing math in pool ctc
  * The initial mint calculation will divide its inputs by the connector unit (10^6 ALGO / 10^18 ETH) and then scale it back. This adds the constraint that initial deposit values may not be fractional. Any idea if this can be avoided?
* Have Announcer announce `Bytes` instead of `Address` to work with ALGO.
* Add conditionals in the frontend for dealing with differences between ALGO and ETH.

### CODE REVIEW

File: examples/duoswap/announcer.mjs [hunk 1]
```diff
@@ -48,8 +50,10 @@ export const runListener = async (useTestnet) => {
   let accListener;
   if (useTestnet) {
     stdlib.setProviderByName('TestNet');
-    const secret = await ask.ask(`What is your secret key?`);
-    accListener = await stdlib.newAccountFromSecret(secret);
+    const secret = await ask.ask(`What is your secret key (ETH) / mnemonic (ALGO) ?`);
```
Comment (jeapostrophe) [inline examples/duoswap/announcer.mjs:53]: DRY

Comment (jeapostrophe) [reply examples/duoswap/announcer.mjs:53]: AND, you know what the connector is, so you know what to ask

**Comment by chrisnevers** [comment]
@jeapostrophe I was just testing with the info I had for my testnet accounts. It was not my goal to make the frontends  complete, but I will add support for those options

**Comment by jeapostrophe** [comment]
I don't think you should stress about... just curious. Maybe it would be easy for you to find the other thing in your testnet accounts.

**Comment by chrisnevers** [comment]
@jeapostrophe I'm running into an error that could possibly be an issue with the ALGO compiler. The program runs on ETH okay. However on ALGO, there's a logic eval error for comparing a `[]byte` with an `uint`. I would expect any errors with my program to occur either statically or during CBR.

The issue occurs when making a trade interactively on a `<network, non-network>` pool, it does not matter whether the `amtInTok` is network or non-network.

https://github.com/reach-sh/reach-lang/blob/97aa51c133de1dcab2284f5b269d9385649a4e2d/examples/duoswap/interactive.mjs#L255-L270

Any thoughts?

```
2021-08-16T21:08:31.426Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- FAIL:
no req, but ["status","response"]
{"status":400,"response":{"req":{"method":"POST","url":"http://devnet-algo:4180/v2/transactions","data":{"type":"Buffer","data":[130,163,115,105,103,196,64,151,74,199,5,139,86,84,24,76,133,212,6,124,228,115,247,109,37,133,1,203,51,80,96,92,12,150,62,121,17,175,172,76,243,72,148,2,116,102,176,88,58,75,105,41,242,221,247,165,65,85,121,37,222,19,111,217,115,244,152,221,57,136,8,163,116,120,110,138,163,97,109,116,206,0,15,66,64,162,102,118,205,18,213,163,103,101,110,169,100,101,118,110,101,116,45,118,49,162,103,104,196,32,242,208,179,199,106,131,182,12,42,221,83,102,195,168,8,87,24,79,4,192,145,104,93,23,187,199,92,45,22,145,199,52,163,103,114,112,196,32,162,15,16,32,161,73,89,24,53,221,46,161,25,248,132,205,36,90,233,39,53,167,114,241,168,229,250,179,104,194,193,222,162,108,118,205,22,189,164,110,111,116,101,196,11,82,101,97,99,104,32,48,46,49,46,51,163,114,99,118,196,32,254,34,206,142,243,234,2,110,139,247,48,187,18,57,63,200,95,163,200,154,164,216,110,173,94,119,50,54,217,214,40,162,163,115,110,100,196,32,116,75,122,248,17,94,170,242,192,210,229,117,149,6,116,223,207,73,58,37,243,53,169,32,49,102,93,245,243,224,219,204,164,116,121,112,101,163,112,97,121,130,164,108,115,105,103,130,163,97,114,103,144,161,108,196,23,4,32,1,1,50,4,34,9,73,56,16,129,6,18,68,56,24,129,201,38,18,68,34,163,116,120,110,139,164,97,97,109,116,206,1,18,27,187,164,97,114,99,118,196,32,116,75,122,248,17,94,170,242,192,210,229,117,149,6,116,223,207,73,58,37,243,53,169,32,49,102,93,245,243,224,219,204,162,102,118,205,18,213,163,103,101,110,169,100,101,118,110,101,116,45,118,49,162,103,104,196,32,242,208,179,199,106,131,182,12,42,221,83,102,195,168,8,87,24,79,4,192,145,104,93,23,187,199,92,45,22,145,199,52,163,103,114,112,196,32,162,15,16,32,161,73,89,24,53,221,46,161,25,248,132,205,36,90,233,39,53,167,114,241,168,229,250,179,104,194,193,222,162,108,118,205,22,189,164,110,111,116,101,196,14,82,101,97,99,104,32,48,46,49,46,51,32,49,41,163,115,110,100,196,32,254,34,206,142,243,234,2,110,139,247,48,187,18,57,63,200,95,163,200,154,164,216,110,173,94,119,50,54,217,214,40,162,164,116,121,112,101,165,97,120,102,101,114,164,120,97,105,100,205,19,1,130,163,115,105,103,196,64,219,104,166,153,80,55,238,0,193,141,183,69,178,219,67,199,219,177,127,111,5,244,167,195,17,84,51,16,201,163,210,50,163,77,176,174,70,95,171,96,59,130,30,13,162,166,161,120,78,23,226,68,53,234,95,114,82,177,212,84,109,201,175,13,163,116,120,110,139,164,97,112,97,97,147,196,1,6,196,97,137,202,47,136,252,158,239,157,107,160,113,115,24,84,254,72,229,192,179,194,24,205,178,209,73,187,2,133,99,213,198,139,0,0,0,0,0,0,19,1,0,0,0,0,0,0,19,81,0,0,98,57,181,162,192,0,0,0,0,0,0,152,150,128,0,0,0,0,0,30,132,128,0,0,0,0,3,55,249,128,255,255,255,255,255,103,105,127,0,0,0,0,0,0,0,22,176,196,42,3,0,0,0,0,0,15,66,64,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,18,27,187,0,0,98,82,225,141,177,192,164,97,112,105,100,205,19,73,163,102,101,101,205,11,184,162,102,118,205,18,213,163,103,101,110,169,100,101,118,110,101,116,45,118,49,162,103,104,196,32,242,208,179,199,106,131,182,12,42,221,83,102,195,168,8,87,24,79,4,192,145,104,93,23,187,199,92,45,22,145,199,52,163,103,114,112,196,32,162,15,16,32,161,73,89,24,53,221,46,161,25,248,132,205,36,90,233,39,53,167,114,241,168,229,250,179,104,194,193,222,162,108,118,205,22,189,164,110,111,116,101,196,11,82,101,97,99,104,32,48,46,49,46,51,163,115,110,100,196,32,116,75,122,248,17,94,170,242,192,210,229,117,149,6,116,223,207,73,58,37,243,53,169,32,49,102,93,245,243,224,219,204,164,116,121,112,101,164,97,112,112,108]},"headers":{"x-algo-api-token":"c87f5580d7a866317b4bfe9e8b8d1dda955636ccebfa88c12b414db208dd9705","content-type":"application/x-binary"}},"header":{"access-control-allow-origin":"*","content-type":"application/json; charset=UTF-8","vary":"Origin","date":"Mon, 16 Aug 2021 21:08:31 GMT","content-length":"158","connection":"close"},"status":400,"text":"{\"message\":\"TransactionPool.Remember: transaction Q6BPHLUASXT4XQB4M76KZS35XRPJYCTN2KAL3TPU627XIJBTV2WA: logic eval error: cannot compare ([]byte to uint64)\"}\n"}}
```

Here is whole log:

<details>

```
Do you want to trade? (y/n)
y
What token do you want to input?
ALGO
CHI
ALGO
How much do you want to trade? (You have 9998.998 ALGO)
1
2021-08-16T21:08:31.375Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- START
2021-08-16T21:08:31.379Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- SIMULATE {
  txns: [
    {
      amt: BigNumber { _hex: '0x0f4240', _isBigNumber: true },
      kind: 'to',
      tok: undefined
    },
    {
      amt: BigNumber { _hex: '0x00', _isBigNumber: true },
      kind: 'to',
      tok: BigNumber { _hex: '0x1301', _isBigNumber: true }
    },
    {
      amt: BigNumber { _hex: '0x00', _isBigNumber: true },
      kind: 'to',
      tok: 4945
    },
    {
      amt: BigNumber { _hex: '0x01121bbb', _isBigNumber: true },
      kind: 'from',
      to: '0x744b7af8115eaaf2c0d2e575950674dfcf493a25f335a92031665df5f3e0dbcc',
      tok: BigNumber { _hex: '0x1301', _isBigNumber: true }
    }
  ],
  mapRefs: [],
  mapsPrev: [],
  mapsNext: [],
  isHalt: false
}
2021-08-16T21:08:31.380Z DEBUG: FXV6: Trader sendrecv m6 time,5808 MAP { mapAccts: [] }
2021-08-16T21:08:31.381Z DEBUG: fillTxn: getting params
2021-08-16T21:08:31.385Z DEBUG: fillTxn: got params: {
  flatFee: false,
  fee: 0,
  firstRound: 4821,
  lastRound: 5821,
  genesisID: 'devnet-v1',
  genesisHash: '8tCzx2qDtgwq3VNmw6gIVxhPBMCRaF0Xu8dcLRaRxzQ='
}
2021-08-16T21:08:31.386Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- TIMECHECK {
  params: {
    flatFee: false,
    fee: 0,
    firstRound: 4821,
    lastRound: 5821,
    genesisID: 'devnet-v1',
    genesisHash: '8tCzx2qDtgwq3VNmw6gIVxhPBMCRaF0Xu8dcLRaRxzQ='
  },
  timeoutAt: [ 'time', BigNumber { _hex: '0x16b0', _isBigNumber: true } ]
}
2021-08-16T21:08:31.387Z DEBUG: checkTimeout {
  timeoutAt: [ 'time', BigNumber { _hex: '0x16b0', _isBigNumber: true } ],
  nowTimeN: 4822
}
2021-08-16T21:08:31.388Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- ASSEMBLE w/ {
  flatFee: false,
  fee: 0,
  firstRound: 4821,
  lastRound: 5821,
  genesisID: 'devnet-v1',
  genesisHash: '8tCzx2qDtgwq3VNmw6gIVxhPBMCRaF0Xu8dcLRaRxzQ='
}
2021-08-16T21:08:31.392Z DEBUG: FXV6: Trader sendrecv m6 time,5808 txnExtraTxns [
  Transaction {
    name: 'Transaction',
    tag: <Buffer 54 58>,
    from: {
      publicKey: Uint8Array(32) [
        116,  75, 122, 248,  17,  94, 170, 242,
        192, 210, 229, 117, 149,   6, 116, 223,
        207,  73,  58,  37, 243,  53, 169,  32,
         49, 102,  93, 245, 243, 224, 219, 204
      ],
      checksum: Uint8Array(4) [ 227, 88, 246, 73 ]
    },
    to: {
      publicKey: Uint8Array(32) [
        254,  34, 206, 142, 243, 234,   2, 110,
        139, 247,  48, 187,  18,  57,  63, 200,
         95, 163, 200, 154, 164, 216, 110, 173,
         94, 119,  50,  54, 217, 214,  40, 162
      ],
      checksum: Uint8Array(4) [ 219, 228, 178, 159 ]
    },
    amount: 1000000n,
    note: Uint8Array(11) [
      82, 101, 97, 99, 104,
      32,  48, 46, 49,  46,
      51
    ],
    type: 'pay',
    flatFee: false,
    genesisHash: <Buffer f2 d0 b3 c7 6a 83 b6 0c 2a dd 53 66 c3 a8 08 57 18 4f 04 c0 91 68 5d 17 bb c7 5c 2d 16 91 c7 34>,
    fee: 0,
    firstRound: 4821,
    lastRound: 5821,
    genesisID: 'devnet-v1',
    appArgs: [],
    lease: Uint8Array(0) [],
    group: undefined
  },
  Transaction {
    name: 'Transaction',
    tag: <Buffer 54 58>,
    type: 'axfer',
    from: {
      publicKey: Uint8Array(32) [
        254,  34, 206, 142, 243, 234,   2, 110,
        139, 247,  48, 187,  18,  57,  63, 200,
         95, 163, 200, 154, 164, 216, 110, 173,
         94, 119,  50,  54, 217, 214,  40, 162
      ],
      checksum: Uint8Array(4) [ 219, 228, 178, 159 ]
    },
    to: {
      publicKey: Uint8Array(32) [
        116,  75, 122, 248,  17,  94, 170, 242,
        192, 210, 229, 117, 149,   6, 116, 223,
        207,  73,  58,  37, 243,  53, 169,  32,
         49, 102,  93, 245, 243, 224, 219, 204
      ],
      checksum: Uint8Array(4) [ 227, 88, 246, 73 ]
    },
    amount: 17963963n,
    assetIndex: 4865,
    note: Uint8Array(14) [
      82, 101, 97, 99, 104, 32,
      48,  46, 49, 46,  51, 32,
      49,  41
    ],
    flatFee: false,
    genesisHash: <Buffer f2 d0 b3 c7 6a 83 b6 0c 2a dd 53 66 c3 a8 08 57 18 4f 04 c0 91 68 5d 17 bb c7 5c 2d 16 91 c7 34>,
    fee: 0,
    firstRound: 4821,
    lastRound: 5821,
    genesisID: 'devnet-v1',
    appArgs: [],
    lease: Uint8Array(0) [],
    group: undefined
  }
]
2021-08-16T21:08:31.392Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- extraFee = 2000
2021-08-16T21:08:31.393Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- ARGS = [
  [
    '0x89ca2f88fc9eef9d6ba071731854fe48e5c0b3c218cdb2d149bb028563d5c68b',
    BigNumber { _hex: '0x1301', _isBigNumber: true },
    4945,
    { k: BigNumber { _hex: '0x6239b5a2c000', _isBigNumber: true } },
    BigNumber { _hex: '0x989680', _isBigNumber: true },
    BigNumber { _hex: '0x1e8480', _isBigNumber: true },
    BigNumber { _hex: '0x0337f980', _isBigNumber: true },
    BigNumber { _hex: '0xffffffffff67697f', _isBigNumber: true },
    false,
    BigNumber { _hex: '0x16b0', _isBigNumber: true }
  ],
  [
    [
      'Trader0',
      {
        amtA: BigNumber { _hex: '0x0f4240', _isBigNumber: true },
        amtB: BigNumber { _hex: '0x00', _isBigNumber: true },
        amtInTok: [ 'None', null ],
        amtOut: BigNumber { _hex: '0x01121bbb', _isBigNumber: true },
        calcK: BigNumber { _hex: '0x6252e18db1c0', _isBigNumber: true }
      }
    ]
  ]
]
2021-08-16T21:08:31.395Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- PREPARE: [
  '06',
  '89ca2f88fc9eef9d6ba071731854fe48e5c0b3c218cdb2d149bb028563d5c68b0000000000001301000000000000135100006239b5a2c000000000000098968000000000001e8480000000000337f980ffffffffff67697f0000000000000016b0',
  '0300000000000f424000000000000000000000000000000000000000000001121bbb00006252e18db1c0'
]
2021-08-16T21:08:31.405Z DEBUG: signed sk_ret
2021-08-16T21:08:31.405Z DEBUG: { txID: 'Q6BPHLUASXT4XQB4M76KZS35XRPJYCTN2KAL3TPU627XIJBTV2WA' }
2021-08-16T21:08:31.408Z DEBUG: {
  sig: Uint8Array(64) [
    219, 104, 166, 153,  80,  55, 238,   0, 193, 141, 183,
     69, 178, 219,  67, 199, 219, 177, 127, 111,   5, 244,
    167, 195,  17,  84,  51,  16, 201, 163, 210,  50, 163,
     77, 176, 174,  70,  95, 171,  96,  59, 130,  30,  13,
    162, 166, 161, 120,  78,  23, 226,  68,  53, 234,  95,
    114,  82, 177, 212,  84, 109, 201, 175,  13
  ],
  txn: {
    apaa: [
      Uint8Array(1) [ 6 ],
      Uint8Array(97) [
        137, 202,  47, 136, 252, 158, 239, 157, 107, 160, 113, 115,
         24,  84, 254,  72, 229, 192, 179, 194,  24, 205, 178, 209,
         73, 187,   2, 133,  99, 213, 198, 139,   0,   0,   0,   0,
          0,   0,  19,   1,   0,   0,   0,   0,   0,   0,  19,  81,
          0,   0,  98,  57, 181, 162, 192,   0,   0,   0,   0,   0,
          0, 152, 150, 128,   0,   0,   0,   0,   0,  30, 132, 128,
          0,   0,   0,   0,   3,  55, 249, 128, 255, 255, 255, 255,
        255, 103, 105, 127,   0,   0,   0,   0,   0,   0,   0,  22,
        176
      ],
      Uint8Array(42) [
          3,   0,  0,   0, 0, 0, 15, 66,  64,   0,
          0,   0,  0,   0, 0, 0,  0,  0,   0,   0,
          0,   0,  0,   0, 0, 0,  0,  0,   0,   0,
          1,  18, 27, 187, 0, 0, 98, 82, 225, 141,
        177, 192
      ]
    ],
    apid: 4937,
    fee: 3000,
    fv: 4821,
    gen: 'devnet-v1',
    gh: Uint8Array(32) [
      242, 208, 179, 199, 106, 131, 182, 12,
       42, 221,  83, 102, 195, 168,   8, 87,
       24,  79,   4, 192, 145, 104,  93, 23,
      187, 199,  92,  45,  22, 145, 199, 52
    ],
    grp: Uint8Array(32) [
      162,  15,  16,  32, 161,  73,  89,  24,
       53, 221,  46, 161,  25, 248, 132, 205,
       36,  90, 233,  39,  53, 167, 114, 241,
      168, 229, 250, 179, 104, 194, 193, 222
    ],
    lv: 5821,
    note: Uint8Array(11) [
      82, 101, 97, 99, 104,
      32,  48, 46, 49,  46,
      51
    ],
    snd: Uint8Array(32) [
      116,  75, 122, 248,  17,  94, 170, 242,
      192, 210, 229, 117, 149,   6, 116, 223,
      207,  73,  58,  37, 243,  53, 169,  32,
       49, 102,  93, 245, 243, 224, 219, 204
    ],
    type: 'appl'
  }
}
2021-08-16T21:08:31.417Z DEBUG: signed sk_ret
2021-08-16T21:08:31.417Z DEBUG: { txID: 'CA3EMSBEC3VUFSV7MH4YUHUH56JTFBVNALMK3AEY7XVA46NSVQSA' }
2021-08-16T21:08:31.418Z DEBUG: {
  sig: Uint8Array(64) [
    151,  74, 199,   5, 139,  86,  84,  24,  76, 133, 212,
      6, 124, 228, 115, 247, 109,  37, 133,   1, 203,  51,
     80,  96,  92,  12, 150,  62, 121,  17, 175, 172,  76,
    243,  72, 148,   2, 116, 102, 176,  88,  58,  75, 105,
     41, 242, 221, 247, 165,  65,  85, 121,  37, 222,  19,
    111, 217, 115, 244, 152, 221,  57, 136,   8
  ],
  txn: {
    amt: 1000000,
    fv: 4821,
    gen: 'devnet-v1',
    gh: Uint8Array(32) [
      242, 208, 179, 199, 106, 131, 182, 12,
       42, 221,  83, 102, 195, 168,   8, 87,
       24,  79,   4, 192, 145, 104,  93, 23,
      187, 199,  92,  45,  22, 145, 199, 52
    ],
    grp: Uint8Array(32) [
      162,  15,  16,  32, 161,  73,  89,  24,
       53, 221,  46, 161,  25, 248, 132, 205,
       36,  90, 233,  39,  53, 167, 114, 241,
      168, 229, 250, 179, 104, 194, 193, 222
    ],
    lv: 5821,
    note: Uint8Array(11) [
      82, 101, 97, 99, 104,
      32,  48, 46, 49,  46,
      51
    ],
    rcv: Uint8Array(32) [
      254,  34, 206, 142, 243, 234,   2, 110,
      139, 247,  48, 187,  18,  57,  63, 200,
       95, 163, 200, 154, 164, 216, 110, 173,
       94, 119,  50,  54, 217, 214,  40, 162
    ],
    snd: Uint8Array(32) [
      116,  75, 122, 248,  17,  94, 170, 242,
      192, 210, 229, 117, 149,   6, 116, 223,
      207,  73,  58,  37, 243,  53, 169,  32,
       49, 102,  93, 245, 243, 224, 219, 204
    ],
    type: 'pay'
  }
}
2021-08-16T21:08:31.419Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- SEND: 3
2021-08-16T21:08:31.420Z DEBUG: sendAndConfirm: gqNzaWfEQJdKxwWLVlQYTIXUBnzkc/dtJYUByzNQYFwMlj55Ea+sTPNIlAJ0ZrBYOktpKfLd96VBVXkl3hNv2XP0mN05iAijdHhuiqNhbXTOAA9CQKJmds0S1aNnZW6pZGV2bmV0LXYxomdoxCDy0LPHaoO2DCrdU2bDqAhXGE8EwJFoXRe7x1wtFpHHNKNncnDEIKIPECChSVkYNd0uoRn4hM0kWuknNady8ajl+rNowsHeomx2zRa9pG5vdGXEC1JlYWNoIDAuMS4zo3JjdsQg/iLOjvPqAm6L9zC7Ejk/yF+jyJqk2G6tXncyNtnWKKKjc25kxCB0S3r4EV6q8sDS5XWVBnTfz0k6JfM1qSAxZl318+DbzKR0eXBlo3BheYKkbHNpZ4KjYXJnkKFsxBcEIAEBMgQiCUk4EIEGEkQ4GIHJJhJEIqN0eG6LpGFhbXTOARIbu6RhcmN2xCB0S3r4EV6q8sDS5XWVBnTfz0k6JfM1qSAxZl318+DbzKJmds0S1aNnZW6pZGV2bmV0LXYxomdoxCDy0LPHaoO2DCrdU2bDqAhXGE8EwJFoXRe7x1wtFpHHNKNncnDEIKIPECChSVkYNd0uoRn4hM0kWuknNady8ajl+rNowsHeomx2zRa9pG5vdGXEDlJlYWNoIDAuMS4zIDEpo3NuZMQg/iLOjvPqAm6L9zC7Ejk/yF+jyJqk2G6tXncyNtnWKKKkdHlwZaVheGZlcqR4YWlkzRMBgqNzaWfEQNtopplQN+4AwY23RbLbQ8fbsX9vBfSnwxFUMxDJo9Iyo02wrkZfq2A7gh4NoqaheE4X4kQ16l9yUrHUVG3Jrw2jdHhui6RhcGFhk8QBBsRhicoviPye751roHFzGFT+SOXAs8IYzbLRSbsChWPVxosAAAAAAAATAQAAAAAAABNRAABiObWiwAAAAAAAAJiWgAAAAAAAHoSAAAAAAAM3+YD//////2dpfwAAAAAAAAAWsMQqAwAAAAAAD0JAAAAAAAAAAAAAAAAAAAAAAAAAAAAAARIbuwAAYlLhjbHApGFwaWTNE0mjZmVlzQu4omZ2zRLVo2dlbqlkZXZuZXQtdjGiZ2jEIPLQs8dqg7YMKt1TZsOoCFcYTwTAkWhdF7vHXC0Wkcc0o2dycMQgog8QIKFJWRg13S6hGfiEzSRa6Sc1p3LxqOX6s2jCwd6ibHbNFr2kbm90ZcQLUmVhY2ggMC4xLjOjc25kxCB0S3r4EV6q8sDS5XWVBnTfz0k6JfM1qSAxZl318+DbzKR0eXBlpGFwcGw=
2021-08-16T21:08:31.426Z DEBUG: FXV6: Trader sendrecv m6 time,5808 --- FAIL:
no req, but ["status","response"]
{"status":400,"response":{"req":{"method":"POST","url":"http://devnet-algo:4180/v2/transactions","data":{"type":"Buffer","data":[130,163,115,105,103,196,64,151,74,199,5,139,86,84,24,76,133,212,6,124,228,115,247,109,37,133,1,203,51,80,96,92,12,150,62,121,17,175,172,76,243,72,148,2,116,102,176,88,58,75,105,41,242,221,247,165,65,85,121,37,222,19,111,217,115,244,152,221,57,136,8,163,116,120,110,138,163,97,109,116,206,0,15,66,64,162,102,118,205,18,213,163,103,101,110,169,100,101,118,110,101,116,45,118,49,162,103,104,196,32,242,208,179,199,106,131,182,12,42,221,83,102,195,168,8,87,24,79,4,192,145,104,93,23,187,199,92,45,22,145,199,52,163,103,114,112,196,32,162,15,16,32,161,73,89,24,53,221,46,161,25,248,132,205,36,90,233,39,53,167,114,241,168,229,250,179,104,194,193,222,162,108,118,205,22,189,164,110,111,116,101,196,11,82,101,97,99,104,32,48,46,49,46,51,163,114,99,118,196,32,254,34,206,142,243,234,2,110,139,247,48,187,18,57,63,200,95,163,200,154,164,216,110,173,94,119,50,54,217,214,40,162,163,115,110,100,196,32,116,75,122,248,17,94,170,242,192,210,229,117,149,6,116,223,207,73,58,37,243,53,169,32,49,102,93,245,243,224,219,204,164,116,121,112,101,163,112,97,121,130,164,108,115,105,103,130,163,97,114,103,144,161,108,196,23,4,32,1,1,50,4,34,9,73,56,16,129,6,18,68,56,24,129,201,38,18,68,34,163,116,120,110,139,164,97,97,109,116,206,1,18,27,187,164,97,114,99,118,196,32,116,75,122,248,17,94,170,242,192,210,229,117,149,6,116,223,207,73,58,37,243,53,169,32,49,102,93,245,243,224,219,204,162,102,118,205,18,213,163,103,101,110,169,100,101,118,110,101,116,45,118,49,162,103,104,196,32,242,208,179,199,106,131,182,12,42,221,83,102,195,168,8,87,24,79,4,192,145,104,93,23,187,199,92,45,22,145,199,52,163,103,114,112,196,32,162,15,16,32,161,73,89,24,53,221,46,161,25,248,132,205,36,90,233,39,53,167,114,241,168,229,250,179,104,194,193,222,162,108,118,205,22,189,164,110,111,116,101,196,14,82,101,97,99,104,32,48,46,49,46,51,32,49,41,163,115,110,100,196,32,254,34,206,142,243,234,2,110,139,247,48,187,18,57,63,200,95,163,200,154,164,216,110,173,94,119,50,54,217,214,40,162,164,116,121,112,101,165,97,120,102,101,114,164,120,97,105,100,205,19,1,130,163,115,105,103,196,64,219,104,166,153,80,55,238,0,193,141,183,69,178,219,67,199,219,177,127,111,5,244,167,195,17,84,51,16,201,163,210,50,163,77,176,174,70,95,171,96,59,130,30,13,162,166,161,120,78,23,226,68,53,234,95,114,82,177,212,84,109,201,175,13,163,116,120,110,139,164,97,112,97,97,147,196,1,6,196,97,137,202,47,136,252,158,239,157,107,160,113,115,24,84,254,72,229,192,179,194,24,205,178,209,73,187,2,133,99,213,198,139,0,0,0,0,0,0,19,1,0,0,0,0,0,0,19,81,0,0,98,57,181,162,192,0,0,0,0,0,0,152,150,128,0,0,0,0,0,30,132,128,0,0,0,0,3,55,249,128,255,255,255,255,255,103,105,127,0,0,0,0,0,0,0,22,176,196,42,3,0,0,0,0,0,15,66,64,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,18,27,187,0,0,98,82,225,141,177,192,164,97,112,105,100,205,19,73,163,102,101,101,205,11,184,162,102,118,205,18,213,163,103,101,110,169,100,101,118,110,101,116,45,118,49,162,103,104,196,32,242,208,179,199,106,131,182,12,42,221,83,102,195,168,8,87,24,79,4,192,145,104,93,23,187,199,92,45,22,145,199,52,163,103,114,112,196,32,162,15,16,32,161,73,89,24,53,221,46,161,25,248,132,205,36,90,233,39,53,167,114,241,168,229,250,179,104,194,193,222,162,108,118,205,22,189,164,110,111,116,101,196,11,82,101,97,99,104,32,48,46,49,46,51,163,115,110,100,196,32,116,75,122,248,17,94,170,242,192,210,229,117,149,6,116,223,207,73,58,37,243,53,169,32,49,102,93,245,243,224,219,204,164,116,121,112,101,164,97,112,112,108]},"headers":{"x-algo-api-token":"c87f5580d7a866317b4bfe9e8b8d1dda955636ccebfa88c12b414db208dd9705","content-type":"application/x-binary"}},"header":{"access-control-allow-origin":"*","content-type":"application/json; charset=UTF-8","vary":"Origin","date":"Mon, 16 Aug 2021 21:08:31 GMT","content-length":"158","connection":"close"},"status":400,"text":"{\"message\":\"TransactionPool.Remember: transaction Q6BPHLUASXT4XQB4M76KZS35XRPJYCTN2KAL3TPU627XIJBTV2WA: logic eval error: cannot compare ([]byte to uint64)\"}\n"}}
2021-08-16T21:08:31.427Z DEBUG: FXV6: Trader recv m6 time,5808 --- START
2021-08-16T21:08:31.427Z DEBUG: FXV6: Trader recv m6 time,5808 EventCache.query {
  ApplicationID: 4937,
  minRound: 4785,
  specRound: undefined,
  timeoutAt: [ 'time', BigNumber { _hex: '0x16b0', _isBigNumber: true } ],
  maxRound: 5808,
  maxSecs: undefined
} 4784
2021-08-16T21:08:31.427Z DEBUG: Transaction not in Event Cache. Querying network...
2021-08-16T21:08:31.428Z DEBUG: FXV6: Trader recv m6 time,5808 --- QUERY = SearchForTransactions {
  c: HTTPClient {
    defaultHeaders: {},
    intDecoding: 'default',
    baseURL: {
      slashes: true,
      protocol: 'http:',
      hash: '',
      query: '',
      pathname: '/',
      auth: '',
      host: 'devnet-algo:8980',
      port: '8980',
      hostname: 'devnet-algo',
      password: '',
      username: '',
      origin: 'http://devnet-algo:8980',
      href: 'http://devnet-algo:8980/'
    },
    tokenHeader: { 'X-Indexer-API-Token': 'reach-devnet' }
  },
  query: { 'application-id': 4937, 'tx-type': 'appl', 'min-round': 4785 },
  intDecoding: 'default'
}
2021-08-16T21:08:31.435Z DEBUG: FXV6: Trader recv m6 time,5808 --- RESULT = { 'current-round': 4821, transactions: [] }
2021-08-16T21:08:31.435Z DEBUG: EventCache res:  { succ: false, round: 4821 }
2021-08-16T21:08:31.435Z DEBUG: checkTimeout {
  timeoutAt: [ 'time', BigNumber { _hex: '0x16b0', _isBigNumber: true } ],
  nowTimeN: 4821
}
^C
```

<details>

**Review by jeapostrophe — APPROVED** [review]

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**